### PR TITLE
eccodes: 2.12.5 -> 2.23.0

### PR DESCRIPTION
**Pull request by matrss**
###### Motivation for this change

Update to eccodes 2.23.0. Perl is now required as a dependency since some version in between.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/eccodes/default.nix [hunk 1]
```diff
@@ -23,6 +23,7 @@ stdenv.mkDerivation rec {
                   openjpeg
                   libpng
                   gfortran
+                  perl
```
Comment (risicle) [inline pkgs/development/libraries/eccodes/default.nix:26]: Is it possible this is just used at build-time and should therefore be in `nativeBuildInputs`?

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Review by SuperSandro2000 — COMMENTED** [review]
Please format the file with nixpkgs-fmt.

**Comment by matrss** [comment]
Moved perl to `nativeBuildInputs` and formatted with `nixpkgs-fmt`, let me know if there is anything else I should do.

File: pkgs/development/libraries/eccodes/default.nix [hunk 2]
```diff
@@ -1,39 +1,52 @@
...
+, perl
+, enablePython ? false
+, pythonPackages
 , enablePosixThreads ? false
-, enableOpenMPThreads ? false}:
+, enableOpenMPThreads ? false
+}:
+
 with lib;
 stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/eccodes/default.nix:17]:
```suggestion
stdenv.mkDerivation rec {
```

Comment (matrss) [reply pkgs/development/libraries/eccodes/default.nix]: It's actually used in `meta` and for `optionals` in `propagatedBuildInputs`, I made the `optionals` explicit and added `with lib;` to meta as seems to be common practice in other packages.

**Pull request merged**